lm_eval/api/model.py

```diff
@@ -247,3 +247,56 @@ def fn(requests):
 
     def get_cache_hook(self):
         return CacheHook(self)
+
+
+class TemplateLM(LM):
+    @property
+    @abc.abstractmethod
+    def eot_token_id(self):
+        pass
+
+    @abc.abstractmethod
+    def tok_encode(self, string: str, add_special_tokens: bool, **kwargs):
+        pass
+
+    @abc.abstractmethod
+    def _loglikelihood_tokens(self, requests, **kwargs):
+        pass
+
+    def _encode_pair(self, context, continuation):
+        n_spaces = len(context) - len(context.rstrip())
+        if n_spaces > 0:
+            continuation = context[-n_spaces:] + continuation
+            context = context[:-n_spaces]
+
+        whole_enc = self.tok_encode(context + continuation, add_special_tokens=False)
```

Comment:
I think this might error out, unless we add `**kwargs` (which then don't get passed to tiktoken) to `tok_encode()` for OpenAICompletionsLM

Reply:
Oh do you mean because of add_special_tokens?

Because I did add kwargs here -- https://github.com/EleutherAI/lm-evaluation-harness/pull/1279/files#diff-0f02af4f2951f7c0eb992894e99368bf0c4cdbd4ef37b9af8a7ae50c1bc08a91R259

But I am guessing you mean to leave out add_special_tokens and instead just do kwargs.

Reply:
yep, looks good now!